edb/server/protocol/binary.pyx

```diff
@@ -1630,15 +1520,58 @@ cdef class EdgeConnection:
             bytes in_tid
             bytes out_tid
             bytes bound_args
+            uint64_t allow_capabilities
+            uint64_t compilation_flags
+            uint64_t implicit_limit
+            bint force_parse
 
-        query, query_req = self.parse_prepare_query_part()
+        self.parse_headers()
+
+        allow_capabilities = <uint64_t>self.buffer.read_int64()
+        compilation_flags = <uint64_t>self.buffer.read_int64()
+        implicit_limit = <uint64_t>self.buffer.read_int64()
+
+        inline_typenames = (
+            compilation_flags
+            & messages.CompilationFlag.INJECT_OUTPUT_TYPE_NAMES
+        )
+        inline_typeids = (
+            compilation_flags
+            & messages.CompilationFlag.INJECT_OUTPUT_TYPE_IDS
+        )
+
+        output_format = self.parse_output_format(self.buffer.read_byte())
+        expect_one = (
+            self.parse_cardinality(self.buffer.read_byte()) is CARD_AT_MOST_ONE
+        )
+
+        query = self.buffer.read_len_prefixed_bytes()
+        if not query:
+            raise errors.BinaryProtocolError('empty query')
+
+        query_req = QueryRequestInfo(
+            self._tokenize(query),
+            self.protocol_version,
+            output_format=output_format,
+            expect_one=expect_one,
+            implicit_limit=implicit_limit,
+            inline_typeids=inline_typeids,
+            inline_typenames=inline_typenames,
+            inline_objectids=True,
+            allow_capabilities=allow_capabilities,
+        )
 
         in_tid = self.buffer.read_bytes(16)
         out_tid = self.buffer.read_bytes(16)
         bind_args = self.buffer.read_len_prefixed_bytes()
         self.buffer.finish_message()
 
+        force_parse = (
```

Comment:
This check is redundant to the tid check below, no?

Reply:
We should actually check against the tid in _last_anon_compiled instead of checking against a hardcoded INVALID_TYPE_ID.